nemo/utils/exp_manager.py

```diff
@@ -877,10 +877,16 @@ def nemo_topk_check_previous_run(self):
         else:
             models_to_delete = len(best_k_models) - self.save_top_k
         logging.debug(f'Number of models to delete: {models_to_delete}')
+
+        # If EMA enabled, delete the additional EMA weights
+        ema_enabled = self._has_ema_ckpts(best_k_models)
```

Comment:
`best_k_models` only has Non-EMA checkpoints, therefore, this would always be **FALSE.**
We need to provide the list of all the checkpoints in the folder here.
```
        all_checkpoints = list(path for path in Path(self.dirpath).rglob("*.ckpt"))
        ema_enabled = self._has_ema_ckpts(all_checkpoints)
```